### PR TITLE
Remove unnecessary hardened runtime exceptions on macOS

### PR DESCRIPTION
**Pull request by kthchew**
These are cruft from Back in the Day™ when the (not Prism) launcher was ad-hoc signed, and the OS couldn't tell that the Qt libraries and launcher were signed by the same party. Library validation is [considered quite important](https://developer.apple.com/documentation/bundleresources/entitlements/com.apple.security.cs.disable-library-validation), so re-enabling this security feature is probably a good idea for security of the launcher itself. For convenience, CI will still apply the disable library validation entitlement when an ad-hoc signature is used.

Allowing dyld environment variables was done to support Steam overlay. To prevent a regression where Steam overlay doesn't work anymore, I pass the `STEAM_DYLD_INSERT_LIBRARIES` environment variable into the game as `DYLD_INSERT_LIBRARIES` when it is launched (usually, the game would inherit `DYLD_INSERT_LIBRARIES` from the launcher process, but with these restrictions doing so explicitly is necessary). I tested this without both of these entitlements and Steam overlay appears to work when opening Minecraft.

### CODE REVIEW

**Review by TheKodeToad — APPROVED** [review]
Looks good, but I think we'd need to merge it to see whether it actually works

**Review by Ryex — APPROVED** [review]

**Comment by Trial97** [comment]
Should we backport this?

**Comment by TheKodeToad** [comment]
I don't think it's critical so I'd move it to 10.0 to ensure it's all working correctly

**Review by txtsd — APPROVED** [review]

**Comment by Trial97** [comment]
And it would be nice to have some labels

**Comment by getchoo** [comment]
> Looks good, but I think we'd need to merge it to see whether it actually works

Should probably do that before a merge. CC @sakecode?

**Comment by kthchew** [comment]
> I don't think it's critical so I'd move it to 10.0 to ensure it's all working correctly

Nah, it's just preventative. I'm not currently aware of any practical attacks against Prism Launcher that involve code injection with libraries. Feel free to push it back further if you want to test.

> Should probably do that before a merge.

To test this with library validation enabled you do need to sign with some stable identity (i.e. not an anonymous ad-hoc signature). Might be wrong but IIRC a free Apple Developer account can use an "Apple Development" type of certificate that enables this kind of use but isn't for distribution [1], so you could do the same thing CI would do to test:

```
codesign --sign "Apple Development: YOUR NAME" --deep --force --entitlements "../program_info/App.entitlements" --options runtime "PrismLauncher.app/Contents/MacOS/prismlauncher"
```

[1] You can check by opening Xcode and going to its Settings > Accounts > select your account > Manage Certificates... If an "Apple Development" certificate is listed there then it should work. Otherwise you should be able to generate one from that screen.

**Comment by DioEgizio** [comment]
Won't this break PR artifacts because actions on PRs don't have the secrets?

**Comment by kthchew** [comment]
> Won't this break PR artifacts because actions on PRs don't have the secrets?

`com.apple.security.cs.disable-library-validation` is in `AdhocSignedApp.entitlements`, which I set CI to use with an ad-hoc signature when the secrets aren't available.

**Review by DioEgizio — APPROVED** [review]
Ah didn't see that, then nvm

**Comment by Trial97** [comment]
I will merge it as it has 4 approvals but please edit the labels until 10.0 is launched

**Pull request merged**
